cmd/ci-operator/main.go

```diff
@@ -933,3 +951,70 @@ func eventRecorder(kubeClient *coreclientset.CoreV1Client, namespace string) rec
 	return eventBroadcaster.NewRecorder(
 		templatescheme.Scheme, coreapi.EventSource{Component: namespace})
 }
+
+func createTemplateServiceAndRoutes(client *coreclientset.CoreV1Client, routeClient *routeclientset.Clientset, templates []*templateapi.Template, namespace string) error {
+	for _, template := range templates {
+		s := &coreapi.Service{
+			ObjectMeta: meta.ObjectMeta{
+				Name: template.Name,
+			},
+			Spec: coreapi.ServiceSpec{
+				PublishNotReadyAddresses: true,
+				Selector:                 map[string]string{"ci-operator.openshift.io/template-name": template.Name},
+				Type:                     coreapi.ServiceTypeClusterIP,
+				Ports: []coreapi.ServicePort{
+					{
+						Name:       "http",
+						Protocol:   "TCP",
+						Port:       80,
+						TargetPort: intstr.FromInt(8080),
+					},
+					{
+						Name:       "https",
+						Protocol:   "TCP",
+						Port:       443,
+						TargetPort: intstr.FromInt(8443),
+					}}}}
+
+		if _, err := client.Services(namespace).Create(s); err != nil {
+			return fmt.Errorf("could not create service %s: %v", template.Name, err)
+		}
+
+		// Create 8080 Route
+		if _, err := routeClient.Route().Routes(namespace).Create(&routev1.Route{
```

Comment:
don't throw away the runtime.object - host is substituted on admission so it will be already set in the returned object.